Codeinwp/neve
# Granular Heading Font Family Controls

### Summary

This PR adds font family controls for headings. At default, headings get inherited from the main typography font; the user is free to override it.

**Reviewer note:**
I ensured current Neve website users who don't have override-specific headings would not be affected by this feature in terms of CSS code or dynamic style. To achieve it, I leveraged dynamic styles [Frontend](https://github.com/Codeinwp/neve/blob/8182bc3a79fa53807372ffaa74c80881102223f0/inc/core/front_end.php#L329-L345) Class instead of CSS files. In that case, any CSS will be rendered if there is no specific form selection. That's different in the customizer; in there all of them are loaded to prevent customizer live css issues. Therefore I've used [separate logic](https://github.com/Codeinwp/neve/blob/8182bc3a79fa53807372ffaa74c80881102223f0/inc/core/front_end.php#L334-L337) for customizer

### Will affect visual aspect of the product

YES

### Screenshots 
![Screen Shot 2023-03-10 at 12 49 35](https://user-images.githubusercontent.com/25361626/224283979-d3013fd5-c2b0-49ad-b718-0300600588fe.jpg)

### Test instructions

Note: that issue is completely part of the Neve Free.

1. Install Neve
2. Visit Customizer->Typography->Headings and you'll see every heading has a font family control
3. Normally, no change is expected on the current Neve users until they override heading font family behavior specifically.
4. Play with specific heading (such as h1,h2) font family controls.
5. Make sure your change reflects on customizer, frontend and inheritance mechanism works fine from end to end. expected inheritance mechanism: (specific heading font family < headings font family < boy font family < fallback font family)


Closes https://github.com/Codeinwp/neve-pro-addon/issues/2204, #3876
